### ISSUE TITLE
"Couldn't recognize the image format for file" on big-endian (PowerPC 32-bit)

### ISSUE THREAD

**Issue by wyatt8740** [olokelo/jxlpy#3]
I have compiled libjxl (with libhwy built with SIMD instructions disabled) on my Powerbook G4 running Debian. The library works fine on my little-endian systems (x86, ARM) but fails on the big endian one. The standalone `cjxl` and `djxl` tools *are* working on the powerbook, so I know that my library build is functional.

My error (in MComix3) is `[ImageHandler] ERROR: Could not load pixbuf for page <n>: `GLib.Error('Couldn't recognize the image format for file "<filename>"', 'gdk-pixbuf-error-quark', 3)`

**Comment by olokelo** [comment]
Hi, thanks for opening the issue!

For the last couple of days I've been trying to compile libjxl on PowerPC 32-bit using Debian 10 port running QEMU. Sadly I ran into some really weird compilation errors while building it. May I know how did you succeed with that? So far I wasn't able to reproduce your issue :(
Honestly I've never used PowerPC architecture before...

If I were you, I would look in 2 places: [where I assume that the image is native endian](https://github.com/olokelo/jxlpy/blob/58efdbe274030c93d544b375b209ee0f7b57d727/_jxlpy/_jxl.pyx#L920) (this is based on [imagecodecs implementation](https://github.com/cgohlke/imagecodecs/blob/e92cef6c1878f0b69ebbfb33f9bb809eccbdc31e/imagecodecs/_jpegxl.pyx#L232)) and in the Pillow [accept function](https://github.com/olokelo/jxlpy/blob/58efdbe274030c93d544b375b209ee0f7b57d727/jxlpy/JXLImagePlugin.py#L8).
Sometimes Pillow's behavior is weird so that if there's an error trying to load the image, it would say the image isn't recognized.

If there's a problem with magic word checking, can you please look at the hex view of the image you're trying to open and see if it's really different than in the _accept function?

**Comment by wyatt8740** [comment]
Yeah, to build libhwy I had to do it with all SIMD support turned off; I also had to comment out the lines in the libjxl makefile relating to some submodule of it that depends on node.js somehow or another - since Blink/V8 still doesn't care about ppc32. It's totally unnecessary to make a working jxl library (tested with `cjxl` and `djxl`), so I'm not sure why it's burned into the makefile.

I'll try to hop on the powerbook sometime soon to get a more concrete description of what I did; I'll also try what you said. I need to sleep now, though.

**Comment by wyatt8740** [comment]
Hey again, quickly before bed I just ran `git diff` on my sources on the powerbook. They're for older git revisions (see the contents of the patch files for the hashes).

These worked on my Debian Sid installation.
[patches_highway_jpegxl.zip](https://github.com/olokelo/jxlpy/files/7029108/patches_highway_jpegxl.zip)

**Comment by olokelo** [comment]
Hi and thanks for patches :)

Lastly I've successfully installed libjxl and jxlpy on my ppc32 Debian. Unfortunately I wasn't able to reproduce your issue.
On my machine everything works fine (tested all scripts in `examples/` directory and decoded some images with no problems).

Can you please try running examples yourself to see if there's a problem with them and if you can, you may provide me the image which your code is failing with?
